packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -83,8 +83,8 @@ export class CoffeeCompiler extends CachingCompiler {
 
     let sourceMap = JSON.parse(output.v3SourceMap);
 
-    if (source.match(/(`|import|export|function\*)/)) {
-      // If source contains backticks or ES2015+ features,
+    if (source.match(/`|\bimport\b|\bexport\b|\byield\b/)) {
```

Comment:
`import` or `export` by definition need to be at the beginning of a line, which in CoffeeScript means they follow a newline (no semicolons). `yield` would usually begin a line after some whitespace; I’m not sure if it could ever not start a line, but it might.

I can’t imagine it that takes more time to run this `match` than it does to recompile with Babel. Assuming most projects don’t use features that generate ES2015 code, which I think is a pretty safe assumption, this regex should be saving us time. And most projects that _do_ generate ES2015 probably are doing so because of modules, which means this regex finds a match almost immediately, from the top of the file, so the `match` should be a very minor performance hit.